src/server/JasmineGraphInstanceService.cpp

```diff
@@ -584,6 +587,271 @@ void *instanceservicesession(void *dummyPt) {
+                    }
+                    instance_logger.log("Sent : Graph ID " + graphID, "info");
+
+                    bzero(data, 301);
+                    read(sockfd, data, 300);
+                    string response = (data);
+                    response = utils.trim_copy(response, " \f\n\r\t\v");
+
+                    if (response.compare(JasmineGraphInstanceProtocol::OK) == 0) {
+                        instance_logger.log("Received : " + JasmineGraphInstanceProtocol::OK, "info");
+                        int partitionID = stoi(workerSocketPair[2]);
+                        result_wr = write(sockfd, std::to_string(partitionID).c_str(), std::to_string(partitionID).size());
+
+                        if (result_wr < 0) {
+                            instance_logger.log("Error writing to socket", "error");
+                        }
+                        instance_logger.log("Sent : Partition ID " + std::to_string(partitionID), "info");
+
+                        string degreeDistString;
+                        while (1) {
+                            bzero(data, 301);
+                            read(sockfd, data, 300);
+                            string response = (data);
+                            response = utils.trim_copy(response, " \f\n\r\t\v");
+
+                            std::string::size_type i = response.find(JasmineGraphInstanceService::END_OF_MESSAGE);
+                            if (i != std::string::npos) {
+                                response.erase(i, JasmineGraphInstanceService::END_OF_MESSAGE.length());
+                                //break when the end of message is received
+                                break;
+                            }
+                            degreeDistString.append(response);
+                        }
+
+                        if (boost::algorithm::ends_with(degreeDistString, ",")) {
+                            degreeDistString.pop_back();
+                        }
+
+                        std::vector<string> workerInDegreeDist;
+                        boost::split(workerInDegreeDist, degreeDistString, boost::is_any_of(","));
+
+                        for (vector<string>::iterator workerInDegreeDistIt=workerInDegreeDist.begin(); workerInDegreeDistIt!=workerInDegreeDist.end(); ++workerInDegreeDistIt) {
+                            std::vector <string> workerInDegreeDistPair;
+
+                            boost::split(workerInDegreeDistPair, *workerInDegreeDistIt, boost::is_any_of(":"));
+                            if (degreeDistribution.count( std::stoi(workerInDegreeDistPair[0]))) {
+
+                                long value = degreeDistribution[std::stoi(workerInDegreeDistPair[0])];
+                                long totalValue = std::stoi(workerInDegreeDistPair[1]) + value;
+
+                                degreeDistribution[std::stoi(workerInDegreeDistPair[0])] = totalValue;
```

Comment:
You can assign the value obtained by `std::stoi(workerInDegreeDistPair[0])]` to a variable and use that variable instead.

Reply:
Fixed